### ISSUE TITLE
contracts on modules don't get checked

### ISSUE THREAD

**Issue by egonSchiele** [egonSchiele/contracts.ruby#44]
See [this issue](https://github.com/snowplow/snowplow/issues/906). Replicate with [this code](https://gist.github.com/egonSchiele/c7d0f44d1e3b2407743e). If you change `module Cli` to `class Cli`, it works.

**Comment by waterlink** [comment]
Hmm, looked into this a bit, and actually modules has `method_added` hook and it works alright.

I looked on this `cli.rb` from linked issue, and discovered, that `method_added` doesn't get overridden on `module Cli` because `Contracts` are included not in the module itself, but in global scope, i.e. in `Object` class. All modules does not inherit from `Object`, so contracts don't get hook up.

Then I tried to fire up a small test script and include `Contracts` only in module, and I got weird error, when doing `self.class.Contract` from `Contract` instance method it is not defined. What it defined on is `self.singleton_class`, but it is not unable to being called, because it can't determine its own owner.

Very confusing, but I feel a solution somewhere near, like handling modules just a bit differently.
